File: LeetCode/LeetCode Practice Problems/1637- Widest Vertical Area Between Two Points Containing No Points.cpp
class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        vector<int> a;
        for(auto i : points) a.push_back(i[0]);
        
        sort(a.begin() , a.end());
        int mx = 0;
        for(int i=1;i<a.size();i++) {
            mx = max(mx , a[i] - a[i-1]);
        }
        
        return mx;
    }
};

File: LeetCode/LeetCode Practice Problems/1637- Widest Vertical Area Between Two Points Containing No Points.java
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        Arrays.sort(points , (a,b) -> Integer.compare(a[0], b[0]));
        
        int ans = 0;
        for(int i=0;i<points.length - 1;i++) {
            ans = Math.max(points[i+1][0] - points[i][0] , ans);
        }
        
        return ans;
    }
}
